Repro for ambiguous rules error

illicitonion opened the pull request:
`./pants test tests/python/pants_test/engine:test_play`

Should error about ambiguous rules, but instead:
`Exception: No installed rules can satisfy Select(Greeting) for a root subject of type FullName.`

If you don't register `full_name_to_greeting` (the source of ambiguity), the test passes.

stuhood commented:
Thanks for the repro! The `No installed rules can satisfy` text no longer exists in master (so a rebase should result in a different error), but points to what might be happening here: the rule graph is being successfully built, but is not ending up with a root that provides this capability. Using `--native-engine-visualize-to` and inspecting the resulting `rule_graph.dot` should help debug.

illicitonion commented:
I just rebased and updated this; it now fails with:
```
E       Exception: Rules with errors: 1
E         (Greeting, [FullName], full_name_to_greeting()):
E           Was not usable by any other @rule.
```
which is better, but doesn't point to the reason being that there's a cycle... Probably good enough for now, I guess :)
